src/library/scala/concurrent/impl/ExecutionContextImpl.scala

```diff
@@ -111,16 +114,24 @@ private[concurrent] object ExecutionContextImpl {
     // The thread factory must provide additional threads to support managed blocking.
     val maxExtraThreads = getInt("scala.concurrent.context.maxExtraThreads", "256")
 
+    val interruptible = getBool("scala.concurrent.context.interruptible")
+
     val uncaughtExceptionHandler: Thread.UncaughtExceptionHandler = new Thread.UncaughtExceptionHandler {
-      override def uncaughtException(thread: Thread, cause: Throwable): Unit = reporter(cause)
+      override def uncaughtException(thread: Thread, cause: Throwable): Unit = {
+        if (interruptible) (thread, cause) match {
+          case (fjw: ForkJoinWorkerThread, _: InterruptedException) => fjw.getPool.shutdownNow()
```

Comment:
Can someone please explain how this is desirable behavior, with precedent?

Reply:
@viktorklang it's totally dubious. The use case was a maven plugin to run a `main` and attempt to ensure daemon threads terminate with shutdown-like semantics (without extra code). I don't think that's a thing, but the linked ticket on scala-js says it is a thing they contend with. This PR is in case someone thinks the use case is useful. The DIY code is not onerous, even for casual usage.

Reply:
Yeah, but this PR is what you absolutely don't want to have. :)

Reply:
I just saw the other comment about vanilla FJP.